### ISSUE TITLE
Custom Manager Pages

### ISSUE THREAD

**Issue by ecnerwal1234** [PiranhaCMS/piranha.core.docs#25]
Are there any instructions on the best way to customize the manger? I see how to add to the menu, but are there examples of adding new content types in the admin?

**Comment by traustitj** [comment]
It probably won't be hard to mimic the identity part, but how do you get the manager to find it? How do you add it?

**Comment by tidyui** [comment]
@traustitj Could you elaborate, I don't really understand what you mean 😊

**Comment by traustitj** [comment]
ok, lets say I write some easy name list management system. How do I add config so I can use it inside the management interface? Is there nothing to add in Configure and Configure services in startup.cs? How do I add a menu for the management? The add/edit/list functions. How do I get it to appear?

A small tutorial about just adding 1 string and have a list and adding inside the management system would be greatly beneficial.

**Comment by filipjansson** [comment]
@traustitj In the next version of the templates we have added an template for creating custom Piranha Module. This template contains an empty pages that is added to the manager menu and some permissions, an embedded file provider is also added for recourses.

I hope this will help you out a bit, you can find the sourse here:
[https://github.com/PiranhaCMS/piranha.core.templates/tree/features/template-options/src/module](https://github.com/PiranhaCMS/piranha.core.templates/tree/features/template-options/src/modulel)

**Comment by traustitj** [comment]
@filipjansson Thank you. I don't need to add anything to startup.cs ?

**Comment by filipjansson** [comment]
@traustitj Yes, and what depends on how you want to do it.

But check out the [TemplateModuleExtensions.cs](https://github.com/PiranhaCMS/piranha.core.templates/blob/features/template-options/src/module/TemplateModuleExtensions.cs) class that has extensions methods for both `PiranhaServiceBuilder `and `IServiceCollection`, that do the same thing more or less.

It just gives you the option to write
```
services.AddPiranha(options =>
{
    ...
    options.UseManager();
    options.UseMyModule();
    ...
});
```
Or
```
services.AddMyModule();
```

**Comment by traustitj** [comment]
Hei. I get some errors. I made a clean project, added things as I think they fit, but it probably is just missing a tiny bit.
Also, I can just call any controller I want to? I know I can, just trying stick to best practices.

Here is the project

**Comment by tidyui** [comment]
@filipjansson When the Menu item is created here:

https://github.com/PiranhaCMS/piranha.core.templates/blob/features/template-options/src/module/TemplateModule.cs#L60

It’s referencing the menu group “PiranhaModule”, but I can’t see any code creating it. Won’t this result in a NullReferenceException?

**Comment by tidyui** [comment]
Did some updates to the module template https://github.com/PiranhaCMS/piranha.core.templates/tree/features/template-options/src/module

**Comment by traustitj** [comment]
@tidyui I updated my project based on your changes. But I still get an error on Menu.Items[....
If you could have a look at my project it would really be awesome and could be used for the documentation

**Comment by traustitj** [comment]
I just fixed the bug, the answer is to register a menu item first. The fix is in my repo.
Feel free to use it for helping with documentations

**Comment by filipjansson** [comment]
@tidyui @traustitj That is correct. I copied the codebase from another project of mine, but in that project I added the module under the Content section in the menu.

**Comment by tidyui** [comment]
Yes @traustitj That fix was present in the changes I pushed (https://github.com/PiranhaCMS/piranha.core.templates/blob/features/template-options/src/module/Module.cs#L60). I tested the template on a new project before committing the changes
